Library/Homebrew/cask/cask.rb

```diff
@@ -38,6 +38,16 @@ def self.all
       end.compact
     end
 
+    def self.full_names
+      Tap.flat_map do |tap|
+        next tap.cask_tokens.blank?
+        next tap.cask_tokens if tap.user == "Homebrew"
+
+        name = tap.name
+        tap.cask_tokens.map { |tok| "#{name}/#{tok}" }
+      end.flatten
+    end
```

Comment:
This just mimics the behavior of `Cask.all.map(&:full_name)` but is significantly faster.